### ISSUE TITLE
[SCEV] Handling of 'both must exit' exit limit 

### ISSUE THREAD

**Issue by LebedevRI** [llvm/llvm-project#53329]
As it can be seen in https://github.com/llvm/llvm-project/blob/650fc40b6d8d9a5869b4fca525d5f237b0ee2803/llvm/lib/Analysis/ScalarEvolution.cpp#L8405-L8410
we still don't deal with that case.

Naively, you'd think that it's just `max(a, b)`:
https://godbolt.org/z/bTbz8hjbn
https://alive2.llvm.org/ce/z/E4ju3C

It is more complicated than that, isn't it?

cc @preames @nikic

**Comment by nikic** [comment]
Yeah, it's not that simple. If you replace `<0,+,1> < x && <0,+,1> < y` with `<0,+,1> != x && <0,+,1> != y` instead, then for x != y the loop will not exit at max(x, y) -- it will never exit.
